enclave/core/shm.h

```diff
@@ -6,23 +6,23 @@
 
 #include <openenclave/bits/types.h>
 
-typedef struct _shared_memory_pool
+typedef struct _shared_memory_arena
 {
     /* Buffer holding the shared memory pool */
     uint8_t* buffer;
     size_t capacity;
     size_t used;
-    struct _shared_memory_pool* next;
-} Shared_memory_pool;
+    struct _shared_memory_arena* next;
+} Shared_memory_arena;
 
 bool oe_configure_shm_capacity(size_t cap);
 
-void* oe_shm_malloc(size_t size);
+void* oe_arena_malloc(size_t size);
```

Comment:
>> Clarification: oe_shm_malloc() is responsible for allocation within a predefined arena, not for the arena allocation, correct? 
>>
> oe_shm_malloc implicitly creates the arena if it's not there already

I think part of my original confusion is that the methods don't do a good job of separating operations:
 - **on** the arena: `oe_reserve_arena()`, `oe_unreserve_arena()`, `oe_arena_destroy()`
 - **within** the arena: `oe_arena_malloc()`, `oe_arena_calloc()`, `oe_arena_clear()`

I would suggest the following to clarify:
- Functions managing an arena use the `verb-noun` pattern (standard):
    - `oe_allocate_arena()`/`oe_deallocate_arena()`: I think it should be made clear that the shared memory is not merely reserved as per VirtualAlloc/mmap semantics, but is actually committed for the arena, as evidenced by our use of `oe_host_malloc` in the SGX implementation. 
    - `oe_teardown_arena()`: `destroy` is usually paired with `create` and it's unclear what it means to destroy vs. deallocate an arena. I think `teardown` generally suggests additional work beyond the deallocation (compare `setup` or `initialize` vs. `create` ... `oe_cleanup_arena()` might also be appropriate for the same reasons) 
- Functions for alloc/dealloc in an arena use `arena` as a prefix modifier to the method (pseudoclass model used in the oecrypto methods):
    - `oe_arena_malloc()`, `oe_arena_calloc()` are fine, but the method names strongly suggest the need for an equivalent `oe_arena_free()` method and it's unclear what `oe_arena_clear()` does in relation to that paradigm. I'd suggest changing that to `oe_arena_free_all()` instead, as it expresses the relationship the deallocation relative to the standard malloc/free model. 
    - The lazy allocation of the arena itself via `oe_arena_malloc()` is fine, though it might be worth calling out the expected semantics in comments for the methods.

Reply:
Nice. Will do.